Mathlib/Topology/ContinuousFunction/UniqueCFC.lean

```diff
@@ -0,0 +1,196 @@
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.NormedSpace.Spectrum
+import Mathlib.Topology.ContinuousFunction.FunctionalCalculus
+import Mathlib.Topology.ContinuousFunction.StoneWeierstrass
+
+/-!
+# Uniqueness of the continuous functional calculus
+
+Let `s : Set 𝕜` be compact where `𝕜` is either `ℝ` or `ℂ`. By the Stone-Weierstrass theroem, the
+(star) subalgebra generated by polynomial functions on `s` is dense in `C(s, 𝕜)`. Moreover, this
+star subalgebra is generated by `X : 𝕜[X]` (i.e., `ContinuousMap.restrict s (.id 𝕜)`) alone.
+Consequently, any continuous star `𝕜`-algebra homomorphism with domain `C(s, 𝕜)`, is uniquely
+determined by its value on `X : 𝕜[X]`.
+
+The same is true for `𝕜 := ℝ≥0`, so long as the algebra `A` is an `ℝ`-algebra, which we establish
+by upgrading a map `C((s : Set ℝ≥0), ℝ≥0) →⋆ₐ[ℝ≥0] A` to `C(((↑) '' s : Set ℝ), ℝ) →⋆ₐ[ℝ] A` in
+the natural way, and then applying the uniqueness for `ℝ`-algebra homomorphisms.
+
+This is the reason the `UniqueContinuousFunctionalCalculus` class exists in the first place, as
+opposed to simply appealing directly to Stone-Weierstrass to prove `StarAlgHom.ext_continuousMap`.
+-/
+
+section UniqueUnital
+
+section IsROrC
+
+variable {𝕜 A : Type*} [IsROrC 𝕜]
+
+theorem IsRORC.uniqueContinuousFunctionalCalculus_of_compactSpace_spectrum [TopologicalSpace A]
+    [T2Space A] [Ring A] [StarRing A] [Algebra 𝕜 A] [h : ∀ a : A, CompactSpace (spectrum 𝕜 a)] :
+    UniqueContinuousFunctionalCalculus 𝕜 A where
+  eq_of_continuous_of_map_id s hs φ ψ hφ hψ h :=
+    ContinuousMap.starAlgHom_ext_map_X hφ hψ <| by
+      convert h using 1
+      all_goals exact congr_arg _ (by ext; simp)
+  compactSpace_spectrum := h
+
+instance IsROrC.instUniqueContinuousFunctionalCalculus [NormedRing A] [StarRing A]
+    [NormedAlgebra 𝕜 A] [CompleteSpace A] : UniqueContinuousFunctionalCalculus 𝕜 A :=
+  IsRORC.uniqueContinuousFunctionalCalculus_of_compactSpace_spectrum
+
+end IsROrC
+
+section NNReal
+open NNReal
+
+variable {A : Type*} [TopologicalSpace A] [Ring A] [StarRing A] [Algebra ℝ A] [TopologicalRing A]
+variable {X : Type*} [TopologicalSpace X]
+
+lemma max_neg_zero {α : Type*} [AddCommGroup α] [LinearOrder α] [CovariantClass α α (· + ·) (· ≤ ·)]
+    (a : α) : max (-a) 0 = max a 0 - a := by
+  have := congr(-$(max_zero_sub_eq_self a))
+  rwa [neg_sub, sub_eq_iff_eq_add', ← sub_eq_add_neg] at this
+
+/-- This map sends `f : C(X, ℝ)` to `f ⊔ 0`, bundled as a continuous map `C(X, ℝ≥0)`. -/
+@[pp_dot]
+noncomputable def ContinuousMap.toNNReal (f : C(X, ℝ)) : C(X, ℝ≥0) :=
+  .realToNNReal |>.comp f
+
+@[fun_prop]
+lemma ContinuousMap.continuous_toNNReal : Continuous (ContinuousMap.toNNReal (X := X)) :=
+  ContinuousMap.continuous_comp _
+
+@[simp]
+lemma ContinuousMap.toNNReal_apply (f : C(X, ℝ)) (x : X) : f.toNNReal x = (f x).toNNReal := rfl
+
+lemma ContinuousMap.toNNReal_add (f g : C(X, ℝ)) :
+    (f + g).toNNReal + (-f).toNNReal + (-g).toNNReal =
```

Comment:
Maybe `ContinuousMap.toNNReal_add_neg` or something like that? This current name suggests something more straightforward. Likewise for `toNNReal_mul` below.